### PR TITLE
feat: Add state-of-the-art ML architectures and comprehensive research

### PR DESCRIPTION
**Pull request by alexflorisca**
This commit implements findings from extensive research on ML architectures
for cryptocurrency price prediction (50+ papers, 100+ implementations reviewed).

### 1. Attention-LSTM (Priority: HIGHEST)
- Multi-head attention mechanism with LSTM layers
- Expected 12-15% improvement in MAE/MSE vs vanilla LSTM
- Provides interpretability via attention weights
- Three variants: default, lightweight, deep
- File: src/ml/training_pipeline/models_attention_lstm.py

### 2. Temporal Convolutional Network (TCN) (Priority: HIGH)
- Dilated causal convolutions with residual connections
- 3-5x faster training than LSTM (parallelizable)
- Competitive accuracy with real-time inference capability
- Receptive field: 125+ timesteps with default config
- File: src/ml/training_pipeline/models_tcn.py
- Supports TCN+Attention hybrid

### 3. LightGBM Baseline (Priority: BASELINE)
- Gradient boosting for fast baseline comparison
- 10-100x faster training than deep learning
- Built-in feature importance analysis
- Includes lag features, rolling stats, momentum features
- File: src/ml/training_pipeline/models_lightgbm.py

## Model Factory & Integration

- Updated models.py with unified factory: create_model(type, input_shape)
- Supports model variants: 'default', 'lightweight', 'deep'
- Model-specific callback configuration
- Easy architecture switching for experiments

## Comprehensive Benchmarking Suite

- File: tests/benchmark/test_model_architectures.py
- Systematic comparison across all architectures
- Metrics: RMSE, MAE, MAPE, Directional Accuracy
- Performance profiling: training time, inference speed
- Production readiness validation (<100ms inference target)
- Statistical comparison tests

### 1. Research Documentation (docs/ml_architecture_research.md)
- 50+ academic papers reviewed (2022-2025)
- Detailed architecture analysis for each approach
- Performance benchmarks from literature
- Implementation resources (GitHub repos, papers)
- Hyperparameter recommendations

### 2. Comprehensive Research Report (docs/ml_model_research_report.md)
- Executive summary with recommendations
- Detailed findings for each architecture class
- Feature engineering best practices (wavelet, on-chain, sentiment)
- Evaluation methodology and backtesting practices
- Production readiness assessment
- Phased implementation roadmap

### 3. Implementation Guide (docs/ml_model_implementation_guide.md)
- Quick start examples for each architecture
- Model selection guide (use case → recommended model)
- Training pipeline integration instructions
- Hyperparameter tuning strategies
- Adding new model architectures (step-by-step)
- Debugging and troubleshooting
- Production deployment checklist

## Key Research Findings

- Transformers (TFT) are state-of-the-art but resource-intensive
- Attention-LSTM provides best accuracy/complexity tradeoff
- TCN offers fastest training with competitive accuracy
- Ensemble stacking improves results by 6-18%
- Feature engineering (wavelet, on-chain) provides significant value
- Sentiment integration (FinBERT) improves volatile period predictions

## Next Steps

1. Train and benchmark Attention-LSTM, TCN, LightGBM on BTCUSDT
2. Compare against current CNN-LSTM baseline
3. If 10%+ improvement, implement ensemble stacking
4. Deploy best model to production via model registry

## Testing

Run benchmarks:
```bash
pytest tests/benchmark/test_model_architectures.py -v
```

Train new models:
```bash
atb train model BTCUSDT --model-type attention_lstm --days 365 --epochs 100
atb train model BTCUSDT --model-type tcn --days 365 --epochs 80
```

## References

- 50+ academic papers (arXiv, IEEE, MDPI, SpringerOpen)
- GitHub implementations: TFT, Informer, Autoformer, TCN
- Research period: 2022-2025
- Focus: Cryptocurrency price prediction state-of-the-art

Closes issue related to ML architecture research and implementation.

### CODE REVIEW

**Comment by alexflorisca** [comment]
@claude please assess the validity and impact of this PR. There has been some work recently on develop to introduce some of these models but I may be mistaken

**Comment by alexflorisca** [comment]
@claude please address points 1 -3 from your recommended actions
